buildSrc/src/main/rocker/com/uber/okbuck/template/config/OkbuckAndroidModules.rocker.raw

```diff
@@ -0,0 +1,113 @@
+@args (
+String androidResourceRule,
+String androidLibraryRule,
+)
+
+def get_rule_dep(rule_name):
+    return ':{}'.format(rule_name)
+
+def get_exts_aar_deps(src_deps):
+    return [dep for dep in src_deps if dep.endswith('.aar')]
+
+def src_to_res(str):
+    return str.replace(":src_", ":res_", 1)
+
+def get_libs_res_deps(src_deps):
+    mapped_res_deps = []
+    for dep in src_deps:
+        if (':src_' in dep and not dep.endswith(':src_main')):
+            mapped_res_deps.append(src_to_res(dep))
+    return mapped_res_deps
+
+# Macro that represents a union of all common rules to for an android_library
+def okbuck_android_module(
+        name,
+        language = None,
+        ## common args
+        visibility = None,
+        package=None,
+        ## Manifest related args
+        manifest_main_manifest = None,
+        manifest_min_sdk = None,
+        manifest_target_sdk = None,
+        manifest_version_code=None,
+        manifest_version_name=None,
+        manifest_debuggable=False,
+        manifest_secondary_manifests=None,
```

Comment:
could we make this a struct? https://docs.bazel.build/versions/master/skylark/lib/struct.html

one for manifest & one for resources. would be cleaner I think.

Reply:
nvm, that would boil down to the same thing as before :|